Automattic/simplenote-android
# Fix for email verification screen appears on accounts verified

Fixes #1343, #1353

### Fix
Fixed the bug in which the email verification screen appears even though the account is verified. To verify whether the account was verified or not, we need the `Account` bucket. This bug was related to a race condition in which the application tried to check if the account wasn't verified before the `Account` bucket was synchronized.

To fix the issue, we wait until the account bucket is synchronized using the `OnNetworkChangeListener` listener with `ChangeType = INDEX`. By using this listener, we make sure that we are checking whether the account is verified when the
`Account` bucket is synchronized. 

<img width=400 src="https://user-images.githubusercontent.com/9729923/110639585-8e855380-81b8-11eb-9c2a-815be115de1b.png" />

### Test

**Note:** to facilitate triggering the bug, disable WiFi and allow just 2G connections on your phone. This can help to trigger the race condition.

Try testing with the previous version and this version:

1. Go to login
2. Tap on Login with Email and introduce your credentials
3. If the account is verified, the email verification screen should not be used. 

### Review

This PR would be preferably reviewed by @dmsnell since it requires knowledge of the `Simperium` listeners and logic.

### Release

`RELEASE-NOTES.txt` was updated in dce439e5bf1d95d18cf3566a1213d66aff1619f8 with:
> Fixed email verification screen appears with verified accounts
